allen-cell-animated/nucmorph-colorizer
# refactor: Cleanup for data scripts

Problem
=======
Cleans up data generation scripts for easier use by scientists!

Solution
========
1. Moves most of the file writing logic to a new `data_writer_utils.py` file, with improved docstring and inline commenting.
2. Updates project Python dependencies and documentation.
3. `generate_data.py` and `generate_data_2.py` (renamed to `convert_nucmorph_data.py` and `convert_emt_migration_data.py`) are now greatly simplified and nearly identical. I recommend using VSCode's comparison tool to diff them.

Partly peer programmed with @toloudis !

## Type of change
* This change requires a documentation update

Steps to Verify:
----------------
I have successfully run the whole data pipeline for the `baby_bear` script without crashing, and output images are identical to the original dataset. None of the object IDs have changed, and all data seems to be included correctly.

If you'd like to see the dataset, you can run the client locally and load the dataset from this link: `http://dev-aics-dtp-001.corp.alleninstitute.org/dan-data/colorizer/data/test/new-dataset/baby_bear/`
